lifecycle_multiphase_test.go

```diff
@@ -0,0 +1,282 @@
+// Copyright The OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build coraza.rule.multiphase_evaluation
+
+// Multiphase specific tests
+// Enabling the multiphase evaluation feature, it is expected to check rule variables at the earliest possible phase. These tests are meant
+// to check circumstances in which multiphase evaluation should anticipate some variables evaluation, leading to an earlier action enforcment and
+// therefore dropping the request at the earliest possible phase.
+// It notably permits to avoid coraza-proxy-wasm bypasses by analyzing received data as soon as possible, before streaming it upstream.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/proxytest"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm/types"
+)
+
+func TestLifecycleMultiMatch(t *testing.T) {
```

Comment:
Should it be MultiMatch or Multiphase?

Reply:
Thanks, I knew that sooner or later, I would have inverted the two 😅